dht.go

```diff
@@ -74,7 +74,7 @@ type IpfsDHT struct {
 
 	stripedPutLocks [256]sync.Mutex
 
-	protocols []protocol.ID // DHT protocols
+	protocols, clientProtocols []protocol.ID // DHT protocols
```

Comment:
In the `moveToServerMode` function, please add docs explaining that we are setting stream handlers for both OLD & NEW protocols as we want to respond to peers who speak both protocols.